lib/rewrite/rewrite-expr.c

```diff
@@ -23,6 +23,7 @@
  */
 
 #include "rewrite/rewrite-expr.h"
+#include "filter/filter-expr.h"
```

Comment:
Does this include still necessary? filter_result_tag is commented out in the end.

Reply:
Yes we need this part, we will use: filter_result_tag() later -> I remove the comment sign from there.

Reply:
You shouldn't reuse filter_result_tag() for rewrite and parser expressions, since a parser or rewrite result is conceptually different from a filter result.

I would simply copy-paste the function with a different name in this case. Obviously if it wasn't just one line, I would reuse its implementation, but it is just one line, so copying it is ok.

Reply:
Thanks. I have created parser_result_tag() and rewrite_result_tag() functions.